lunatrace/bsl/ingest-worker/pkg/vulnerability/advisory/osv.go

```diff
@@ -334,13 +350,64 @@ func mapAffectedElement(element schema.OsvSchemaAffectedElem) *gql.Vulnerability
 		On_conflict: VulnerabilityPackageOnConflict,
 	}
 
-	return &gql.Vulnerability_affected_insert_input{
+	queryBody := &gql.Vulnerability_affected_insert_input{
 		Affected_range_events: rangeEvents,
 		Affected_versions:     mapAffectedVersion(element.Versions),
 		Database_specific:     util.Ptr(databaseSpecific),
 		Ecosystem_specific:    util.Ptr(ecosystemSpecific),
 		Package:               affectedPackage,
 	}
+
+	if len(element.Ranges) != 0 {
+		queryBody.Ranges = buildRangeInsert(element.Ranges)
+	}
+
+	return queryBody
+}
+
+func buildRangeInsert(rawRanges []schema.OsvSchemaAffectedElemRangesElem) (rangesInput *gql.Vulnerability_range_arr_rel_insert_input) {
+	ranges := parseRanges(rawRanges)
+	rangesInput = &gql.Vulnerability_range_arr_rel_insert_input{
+		Data:        ranges,
+		On_conflict: RangeOnConflict,
+	}
+	return rangesInput
+}
+
+// This is the replacement for the functions below. This maps the ranges in a flatter way, into a single table
+// For now both sets of functions and tables exist in parallel for legacy reasons, but hopefully this table will be good enough that we can deprecate the other tabes
+func parseRanges(rawRanges []schema.OsvSchemaAffectedElemRangesElem) (ranges []*gql.Vulnerability_range_insert_input) {
+
+	existingRangeSlugs := make(map[string]bool)
+
+	for _, rawRange := range rawRanges {
+		var parsedRange gql.Vulnerability_range_insert_input
+		slug := ""
+		for _, event := range rawRange.Events {
+			for eventPropName, eventPropValue := range event {
+				versionString := getEventValue(eventPropValue)
+				version, err := semver.ParseTolerant(versionString)
+				sanitizedVersionString := version.String()
+				if err != nil {
+					continue
+				}
+				if eventPropName == "introduced" {
+					parsedRange.Introduced = &sanitizedVersionString
+					slug += "introduced:" + sanitizedVersionString
+				}
+				if eventPropName == "fixed" {
+					parsedRange.Fixed = &sanitizedVersionString
+					slug += "fixed:" + sanitizedVersionString
+				}
+			}
+		}
+
+		if existingRangeSlugs[slug] != true {
```

Comment:
If you insert a duplicate that violates the conflict in the SAME query, postgres will throw, so we have to pre check.  Maybe theres a better way

Reply:
this seems like the right thing to do here for now